org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/ConnectionAwareLauncherProxy.java

```diff
@@ -57,62 +57,54 @@ private ConnectionAwareLauncherProxy(ProjectConnection connection, LongRunningOp
+        applyConfiguration(launcher, gradleArguments, buildEnvironment, configWriter, transientAttributes);
         return (BuildLauncher) newProxyInstance(connection, launcher);
     }
 
-    static TestLauncher newTestLauncher(RunConfiguration runConfiguration, BuildEnvironment buildEnvironment, TransientRequestAttributes transientAttributes) {
-        ProjectConnection connection = openConnection(runConfiguration);
+    static TestLauncher newTestLauncher(GradleArguments gradleArguments, Writer configWriter, TransientRequestAttributes transientAttributes) {
+        ProjectConnection connection = openConnection(gradleArguments);
+        BuildEnvironment buildEnvironment = connection.getModel(BuildEnvironment.class);
         TestLauncher launcher = connection.newTestLauncher();
-        applyConfiguration(launcher, runConfiguration, buildEnvironment, transientAttributes);
+        applyConfiguration(launcher, gradleArguments, buildEnvironment, configWriter, transientAttributes);
         return (TestLauncher) newProxyInstance(connection, launcher);
     }
 
-    private static ProjectConnection openConnection(BuildConfiguration buildConfiguration) {
+    private static ProjectConnection openConnection(GradleArguments gradleArguments) {
         GradleConnector connector = GradleConnector.newConnector();
-        buildConfiguration.applyTo(connector);
+        gradleArguments.applyTo(connector);
         return connector.connect();
     }
 
-    private static ProjectConnection openConnection(RunConfiguration runConfiguration) {
-        GradleConnector connector = GradleConnector.newConnector();
-        runConfiguration.applyTo(connector);
-        return connector.connect();
-    }
-
-    private static void applyConfiguration(LongRunningOperation operation, BuildConfiguration buildConfiguration, BuildEnvironment buildEnvironment,
+    private static void applyConfiguration(LongRunningOperation operation, GradleArguments gradleArguments, BuildEnvironment buildEnvironment,
             TransientRequestAttributes transientAttributes) {
-        buildConfiguration.applyTo(operation, buildEnvironment);
+        gradleArguments.applyTo(operation, buildEnvironment);
         applyTransientAttributes(operation, transientAttributes);
     }
 
-    private static void applyConfiguration(LongRunningOperation operation, RunConfiguration runConfiguration, BuildEnvironment buildEnvironment,
-            TransientRequestAttributes transientAttributes) {
-        runConfiguration.applyTo(operation, buildEnvironment);
+    private static void applyConfiguration(LongRunningOperation operation, GradleArguments gradleArguments, BuildEnvironment buildEnvironment,
+            Writer configWriter, TransientRequestAttributes transientAttributes) {
+        gradleArguments.applyTo(operation, buildEnvironment);
+        gradleArguments.describe(configWriter, buildEnvironment);
```

Comment:
I'm a bit confused on this. I moved the `Writer` argument to where it is now because that's where we also moved the `BuildEnvironment` query. 

We can query the `BuildEnvironment` in `BaseLaunchRequest`, but for that, we need to convert the `RunConfiguration` to a `BuildConfiguration`. This is what the `buildConfigurationFor()` method did.